src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -812,6 +813,75 @@ public function simplePaginate($perPage = null, $columns = ['*'], $pageName = 'p
         ]);
     }
 
+    /**
+     * Paginate the given query into a cursor paginator.
+     *
+     * @param  int|null  $perPage
+     * @param  array  $columns
+     * @param  string  $cursorName
+     * @param  string|null  $cursor
+     * @return \Illuminate\Contracts\Pagination\Paginator
+     * @throws \Exception
+     */
+    public function cursorPaginate($perPage = null, $columns = ['*'], $cursorName = 'cursor', $cursor = null)
+    {
+        $cursor = $cursor ?: CursorPaginator::resolveCurrentCursor($cursorName);
+
+        $perPage = $perPage ?: $this->model->getPerPage();
+
+        $orders = $this->ensureOrderForCursorPagination(! is_null($cursor) && $cursor->isPrev());
+
+        $orderDirection = $orders->first()['direction'] ?? 'asc';
+
+        $comparisonOperator = ($orderDirection === 'asc' ? '>' : '<');
+
+        $parameters = $orders->pluck('column')->toArray();
+
+        if (count($parameters) === 1 && ! is_null($cursor)) {
```

Comment:
Perhaps nest inside the `! is_null($cursor)` check, since that applies to both branches?

Reply:
Have included this change, thanks! 👍